assets/js/modules/analytics/datastore/settings.js

```diff
@@ -98,12 +98,13 @@ export async function submitChanges( { select, dispatch } ) {
 	// TODO: Remove once legacy dataAPI is no longer used.
 	invalidateCacheGroup( TYPE_MODULES, 'analytics' );
 
-	if ( isFeatureEnabled( 'ga4setup' ) ) {
-		if ( select( MODULES_ANALYTICS_4 ).haveSettingsChanged() ) {
-			const { error } = await dispatch( MODULES_ANALYTICS_4 ).submitChanges();
-			if ( isPermissionScopeError( error ) ) {
-				return { error };
-			}
+	// Try to save GA4 settings only if the Analytics-4 module has the same connection status as the Analytics module has.
+	// This check is needed to prevent saving GA4 settings unintentionally for cases when the Analytics module has been
+	// already connected by the time the GA4 functionality is revealed.
```

Comment:
This comment is now no longer accurate. It might be okay to just remove it.

Reply:
Removed.